### DISCUSSION TITLE
Does orama support "nested conditions" and "Or" operation in filtering?

### DISCUSSION THREAD

**Discussion by hazeebawan** [oramasearch/orama#670] [Q&A] [answered]
Orama documentation on filtering - https://docs.oramasearch.com/open-source/usage/search/filters - presents this kind of structure for applying filters:

```
const results = await search(db, {
  term: 'prestige',
  where: {
    year: {
      gte: 2000,
    },
    'meta.rating': {
      between: [5, 10],
    },
    'meta.length': {
      lte: 60,
    }
  },
})
````

By the looks of it, this implies that these conditions are joined by an "and", and there is also no talk of an "or" operator or nested conditions. So in short, my question is whether orama support or plan to support this in future?

Basically something like this mongodb filter:

```
status: 'A',
$or: [
    { qty: { $lt: 30 } }, { item: { $regex: '^p' } }
  ]
}
```

or this SQL where clause:

```
WHERE status = "A" AND ( qty < 30 OR item LIKE "p%")
```

**Comment by micheleriva** [comment, marked as answer]
Hi @AHaseebAwan ,
We don't support this right now. Might be in the roadmap for the future, but I don't have an exact timeframe to give you right now.

**Reply by isaachinman** [reply]
@micheleriva Has there been any movement or update on supporting an OR operator? Having to perform two separate searches and combine results is really problematic when also dealing with limit+offset.

**Reply by micheleriva** [reply]
Hi @isaachinman, we have implemented it in OramaCore (https://github.com/oramasearch/oramacore), and adding it to Orama.js is in the plans. I don't have a strict ETA, but we'll try to unify the APIs among the two projects. I'm sorry if that's taking longer than expected

**Reply by isaachinman** [reply]
Ah, that is great to hear! I do not see any mention of an `OR` operator in the [core docs](https://docs.oramacore.com/docs/oramacore/apis/search-documents). Does documentation exist yet?

If I can help with a PR, please let me know.

**Reply by micheleriva** [reply]
We're writing the docs these days! I hope I can send you some by the end of the week, but I can't promise it right now :(
